lib/PuppeteerSharp/JSHandle.cs

```diff
@@ -9,34 +9,25 @@
 
 namespace PuppeteerSharp
 {
-    /// <summary>
-    /// JSHandle represents an in-page JavaScript object. JSHandles can be created with the <see cref="Page.EvaluateExpressionHandleAsync(string)"/> and <see cref="Page.EvaluateFunctionHandleAsync(string, object[])"/> methods.
-    /// </summary>
+    /// <inheritdoc/>
     [JsonConverter(typeof(JSHandleMethodConverter))]
-    public class JSHandle
+    public class JSHandle : IJSHandle
     {
-        internal JSHandle(ExecutionContext context, CDPSession client, RemoteObject remoteObject)
+        private readonly IExecutionContext _executionContext;
+
+        internal JSHandle(IExecutionContext context, CDPSession client, RemoteObject remoteObject)
         {
-            ExecutionContext = context;
+            _executionContext = context;
             Client = client;
             Logger = Client.Connection.LoggerFactory.CreateLogger(GetType());
             RemoteObject = remoteObject;
         }
 
-        /// <summary>
-        /// Gets the execution context.
-        /// </summary>
-        /// <value>The execution context.</value>
-        public ExecutionContext ExecutionContext { get; }
-        /// <summary>
-        /// Gets a value indicating whether this <see cref="JSHandle"/> is disposed.
-        /// </summary>
-        /// <value><c>true</c> if disposed; otherwise, <c>false</c>.</value>
+        /// <inheritdoc/>
+        public IExecutionContext ExecutionContext => _executionContext;
```

Comment:
I think we can keep this as `public IExecutionContext ExecutionContext { get; }` right?

Reply:
One patter I would like to implement is this one:

```
IExecutionContext IJSHandle.ExecutionContext => ExecutionContext;

public ExecutionContext ExecutionContext { get; }
```

In this way if you variable is `JSHandle` instead of `IJSHandle` you would get all the implementations and reduce friction with existing users and remove some casts.
I would apply this across the board.